File: src/device/gyrosensor_subscribe.cpp
#include "ros/ros.h"
#include "silbot3_msgs/Device_Gyro_Msg.h"

void callback(const silbot3_msgs::Device_Gyro_Msg::ConstPtr& msg)
{
  ROS_INFO("angle : %f, rate : %f", msg->angle, msg->rate);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "gyro_listener_cpp");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("/DeviceNode/Gyro/data", 1000, callback);

  ros::spin();

  return 0;
}

File: scripts/device/gyrosensor_subscribe.py
#! /usr/bin/env python
import rospy
from silbot3_msgs.msg import Device_Gyro_Msg

def callback(msg) :
    rospy.loginfo("angle : " + str(msg.angle) +", rate : "+str(msg.rate))

def listen_sonarsensor() :
    subscriber = rospy.Subscriber('/DeviceNode/Gyro/data', Device_Gyro_Msg, callback)

if __name__ == "__main__" :
    rospy.init_node("gyro_listener_python", anonymous=True)
    listen_sonarsensor()
    rospy.spin()
